src/data/transform.ts

```diff
@@ -0,0 +1,555 @@
+  const height = linearElement.height || DEFAULT_LINEAR_ELEMENT_PROPS.height;
+  const excliadrawLinearElement = newLinearElement({
+    type,
+    width,
+    height,
+    points: [
+      [0, 0],
+      [width, height],
+    ],
+    endArrowhead,
+    ...rest,
+  });
+
+  let startBoundElement;
+  let endBoundElement;
+
+  mutateElement(excliadrawLinearElement, {
+    startBinding: linearElement?.startBinding || null,
+    endBinding: linearElement.endBinding || null,
+  });
+
+  if (start) {
+    const width = start?.width ?? DEFAULT_DIMENSION;
+    const height = start?.height ?? DEFAULT_DIMENSION;
+
+    let existingElement;
+    if (start.id) {
+      existingElement = excalidrawElements
+        .get()
+        .find((ele) => ele?.id === start.id) as Exclude<
+        ExcalidrawBindableElement,
+        ExcalidrawImageElement | ExcalidrawFrameElement
+      >;
+      if (!existingElement) {
+        console.error(`No element for start binding with id ${start.id} found`);
+      }
+    }
+
+    const startX = start.x || excliadrawLinearElement.x - width;
+    const startY = start.y || excliadrawLinearElement.y - height / 2;
+    const startType = existingElement ? existingElement.type : start.type;
+
+    if (startType) {
+      if (startType === "text") {
+        let text = "";
+        if (existingElement && existingElement.type === "text") {
+          text = existingElement.text;
+        } else if (start.type === "text") {
+          text = start.text;
+        }
+        if (!text) {
+          console.error(
+            `No text found for start binding text element for ${excliadrawLinearElement.id}`,
+          );
+        }
+        startBoundElement = newTextElement({
+          x: startX,
+          y: startY,
+          type: "text",
+          ...existingElement,
+          ...start,
+          text,
+        });
+        // to position the text correctly when coordinates not provided
+        mutateElement(startBoundElement, {
+          x: start.x || excliadrawLinearElement.x - startBoundElement.width,
+          y:
+            start.y || excliadrawLinearElement.y - startBoundElement.height / 2,
+        });
+      } else {
+        startBoundElement = newElement({
+          x: startX,
+          y: startY,
+          width,
+          height,
+          ...existingElement,
+          ...start,
+          type: startType,
+        });
+      }
+
+      bindLinearElement(
+        excliadrawLinearElement,
+        startBoundElement as ExcalidrawBindableElement,
+        "start",
+      );
+    }
+  }
+  if (end) {
+    const height = end?.height ?? DEFAULT_DIMENSION;
+    const width = end?.width ?? DEFAULT_DIMENSION;
+
+    let existingElement;
+    if (end.id) {
+      existingElement = excalidrawElements
+        .get()
+        .find((ele) => ele?.id === end.id) as Exclude<
+        ExcalidrawBindableElement,
+        ExcalidrawImageElement | ExcalidrawFrameElement
+      >;
```

Comment:
why excluding the image & frame elements here?

Also, let's use `.find((ele): el is AssertedType => /* ... */)` as it's slightly safer (checks  assignability). Applies to assertions below as well.

Reply:
For image I think the API needs to get better which it isn't yet, as user needs to call `addFiles` to make it work. and Frames was added later so I didn't include it yet. I think both of these can be added once we merge this PR.